### PR TITLE
CMake: changes needed to build on Linux

### PR DESCRIPTION
**Pull request by ideasman42**
Note that these are the main changes from the linux port.
Not a fully working linux port.

Merging this will make updating from master less hassle.

### CODE REVIEW

File: toonz/sources/CMakeLists.txt [hunk 1]
```diff
@@ -198,13 +219,65 @@ else()
...
+    find_library(GL_LIB GL)
+    find_library(GLU_LIB GLU)
+    find_library(GLUT_LIB glut)
+    find_library(Z_LIB z)
+
+    find_package(TIFF REQUIRED)
+    set(TIFF_LIB ${TIFF_LIBRARY})
+    # the libraries have not .pc nor preferred Find*.cmake. use custom modules.
+    # find_package(PNG REQUIRED)
+    set(PNG_LIB ${SDKROOT}/libpng-1.6.21/.libs/libpng16.so)
```
Comment (janisozaur) [inline toonz/sources/CMakeLists.txt]: Many distros ship current versions of libpng, can this be changed into a fallback only when system libpng is not found?

Comment (J5lx) [inline toonz/sources/CMakeLists.txt]: If anything, there should be a normal option for it IMO. Autodetecting stuff like that and building a software differently based on the result is bad for packaging that software reproducibly (i.e. the packaging process will result in different packages depending on whether libpng is installed on the build system or not. So if some package maintainer is unaware of the autodetection he might create a package without the libpng dependency and someone with system libpng available will end up with a package with incomplete dependency information).

Comment (ideasman42) [inline toonz/sources/CMakeLists.txt]: Updated it to use systems libpng.

@J5lx, CMake lists libraries used, package maintainers can pass them explicitly to CMake if they don't want to use auto-detection.

Comment (J5lx) [inline toonz/sources/CMakeLists.txt]: Even so I believe it's better to be explicit about dependencies. But I guess it's a somewhat controversial topic after all (though far less controversial than code style or editors :wink:) so let's not make mountains out of molehills. I'll just try to find that list of libraries in the future.

**Comment by skitaoka** [comment]
Jenkins, test this.

**Comment by skitaoka** [comment]
LGTM

File: toonz/sources/CMakeLists.txt (hunk 1 above)
Comment (ideasman42) [inline toonz/sources/CMakeLists.txt]: Being explicit about dependencies within the CMake files can make bisecting older versions of the software difficult. (unless your project keeps an image of the platform used to build each version - some do).
In my experience its best to store exact configurations in a separate CMake file or build script.

**Pull request merged**